api/rest/config.go

```diff
@@ -122,7 +143,21 @@ func (cfg *Config) Validate() error {
 	}
 
 	if (cfg.pathSSLCertFile != "" || cfg.pathSSLKeyFile != "") && cfg.TLS == nil {
-		return errors.New("error loading SSL certificate or key")
+		return errors.New("missing TLS configuration")
+	}
+
+	return cfg.validateLibp2p()
+}
+
+func (cfg *Config) validateLibp2p() error {
+	if cfg.ID != "" || cfg.PrivateKey != nil || cfg.Libp2pListenAddr != nil {
```

Comment:
Because in your proposal it will error if all 3 cfg.ID+privateKey+ListenAddr are NOT set and it should not.

It should only error one or more, but not all, are set.